### ISSUE TITLE
Crash when spamming C-w in an empty file

### ISSUE THREAD

**Issue by VayuDev** [mcobzarenco/zee#16]
To reproduce, just open an empty file and spam C-w, after a few key presses zee crashes with the following message:

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', .../ropey-1.4.1/src/rope.rs:952:36
stack backtrace:
0: rust_begin_unwind at /rustc/1.59.0/library/std/src/panicking.rs:498:5
1: core::panicking::panic_fmt at /rustc/1.59.0/library/core/src/panicking.rs:116:14
2: core::panicking::panic at /rustc/1.59.0/library/core/src/panicking.rs:48:5
3: <zee::components::buffer::Buffer as zee::components::Component>::reduce
4: zee::editor::Editor::handle_event
5: zee::run_editor_ui_loop
6: zee::start_editor
7: zee::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fish: Job 1, 'RUST_BACKTRACE=1 zee test.txt' terminated by signal SIGABRT (Abort)
```

Also, the formatting of the panic is horrible, maybe there should be a custom panic handler that first resets the terminal?

**Comment by mcobzarenco** [comment]
Ooops, thank you for checking zee out and for the bug report -- I just pushed a commit to correctly handle empty buffers when cutting.

> Also, the formatting of the panic is horrible, maybe there should be a custom panic handler that first resets the terminal?

This is a great idea and wondered previously how to get nice panics -- locally I redirect stderr when investigating, e.g. `zee file 2> zee.log`, but even so the terminal stays stuck in raw mode. A custom panic handler may be a neat solution, I'll try it.

**Comment by mcobzarenco** [comment]
As far as I can tell this is fixed by 43b98f5baf34d6f9b6c5404e517c99cfbac3c744 -- I will go ahead and close the issue, please reopen if that's not the case for you

**Comment by mcobzarenco** [comment]
> Also, the formatting of the panic is horrible, maybe there should be a custom panic handler that first resets the terminal?

Just a note for posterity that I added a custom panic handler in https://github.com/zee-editor/zee/commit/d2cae717f2cc2cd68bf68b918fda80d533b96d09 to print the panic after unwinding and changed the release profile from abort -> unwind -- should really unwind to ensure the terminal gets reset from raw mode (and the panic is printed nicely). I do hope it won't be needed, ha smile
